subprojects/core/src/main/java/org/gradle/execution/plan/LocalTaskNode.java

```diff
@@ -180,17 +182,38 @@ public String toString() {
         return task.getIdentityPath().toString();
     }
 
+    @Override
+    public void ensurePropertiesFinalized() {
+        if (!finalized) {
```

Comment:
Could this check for `taskProperties == null` instead of introducing a separate property?